benwbrum/fromthepage
# 1220 clean up deed type strings

Closes #1220 
This PR is dependent on PR #1214  being merged into `development` first.

## Summary
This branch replaces instances of where a deed type is referred to as its string (ex: `page_trans`) instead of its constant (ex: `DeedType::PAGE_TRANSCRIPTION`). It also addresses logic for groups of deed types that are reused with methods like `DeedType.edited_pages` replacing a whole array like `[DeedType::PAGE_EDIT, DeedType::NEEDS_REVIEW, DeedType::OCR_CORRECTED ]`. The goal of these replacements is to reduce the chance of errors when listing a deed type or set of deed types. A refactor plus new testing for the `nightly_user_activity` rake task is also included.

There are 3 instances in the `specs/fixtures/deeds.yml` file that I have not been able to figure out how to convert to their corresponding constant.

## Need for Input
There are a few groups of deed types in the `app/helpers/user_helper.rb` that look like they might be redundant. I'd love some input on what deed types should be used in those places. 

## Testing for Mailers
Since part of these string replacements involved the `nightly_user_activity.rake`, we discovered there was no testing in place for the mailer. I've refactored the mailer and added testing for it. In the process, I've added config to our development environments so that we can use [MailCatcher](https://mailcatcher.me/). MailCatcher captures the emails we generate in the dev env from things like rake tasks and displays them for us in an inbox like this:
<img width="956" alt="screenshot 2018-10-26 06 28 28" src="https://user-images.githubusercontent.com/8680712/47564659-5cc30900-d8eb-11e8-87cf-1c61023cfb49.png">

With MailCatcher, we can confidently work through mailer testing & debugging without worrying that our customers will be getting emails from us. Hooray! The creators of MailCatcher recommend that we do *not* put this gem in our Gemfile, but rather to install it locally. 

1.  `gem install mailcatcher`
2. run `mailcatcher` to start the mail server
3. Go to http://localhost:1080/
4. Send a mailer as you normally would and see it appear in the browser
5. Stop the server with the "Close" button in the browser

## Bug Fixed
See issue #1223 
There is some follow-up work in the production database to be done.

## Recommendation for Similar Work
While doing this work, I noticed that the `Page` model statuses are in a similar condition with their constants being used in some places and their strings being used in other places. This may be worth looking into as well.
